sn_transfers/src/wallet/error.rs

```diff
@@ -8,12 +8,18 @@
 
 use thiserror::Error;
 
+use crate::UniquePubkey;
+
 /// Specialisation of `std::Result`.
 pub type Result<T, E = Error> = std::result::Result<T, E>;
 
 /// Transfer errors.
 #[derive(Debug, Error)]
 pub enum Error {
+    /// The cashnote that was attempted to be spent has already been spent and is not the same
+    #[error("Double spend attempted")]
+    DoubleSpendAttempted(Vec<UniquePubkey>),
```

Comment:
Just to make sure here does the `Vec<UniquePubkey>` contain multiple different double spend attempts? If so I'd add a note in the comments saying this error can contain multiple double spends. Also since the `UniquePubkey` are different, I'd make it a `BTreeSet` so we're sure that there are no duplicates.

Reply:
It doesn't necessarily, just that it already exists. (And _this_ would have been a double spend, but we catch it up front).

I'll; add notes and btreeset to that affect